interview-questions.md

```diff
@@ -8,62 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: In my experience parameters are parenthesis that will hold arguments. arguments can be used to get a certain output from your code.
 
-  Researched answer:
+  Researched answer: A parameter is a variable that is passed into a function. While argument hold a value inside a function.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: map will take three predefined parameters which are value, index and array. Only the value is required while the other two index and array are optional.
 
-  Researched answer:
+  Researched answer: .map has three built in parameters which are value, index, and array. This higher order function has to take at least one argument which is always value, but can take two or three. The other two optional arguments are index and array
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: In my experience map will iterate through an array and return the same array. While filter will iterate through the array, but will return a new array.
 
-  Researched answer:
+  Researched answer: Map will iterate through an array and return a new array of the same length while filter will iterate through an array and return a new array with only the values that are true or complete your condition.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function in my experience will begin your code and be assigned a name. You will use this name to be able to see the output of your code. A method is used to change the output of your code
 
-  Researched answer:
+  Researched answer: A function is a set of instructions on how you want to output your code. While a method is a set of instructions that go with an object.
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: In my experience object destructuring is when you take your object and break it down to return only certain values.
 
-  Researched answer:
+  Researched answer: Object destructuring allows you to take apart an array or object and use their vaules to make new variables.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: In my experience so far in JavaScript I haven't used the word hoisting. If I were to break down the word, hoist is to bring something up.  Hoisting in JavaScript might be be something at the top of the code. I would like to do more research on the word and get back to you.
```

Comment:
Excellent answer!  You won't always know the exact answer when answering at an interview, but relating it to something you can understand and showing interest and curiosity on it will really help you stand out!